src/BackupExplorer/Parser/BackupParser.cs

```diff
@@ -35,6 +35,35 @@ public BackupParser(string backupChainPath)
             this.backupParserImpl = new BackupParserImpl(backupChainPath, String.Empty);
         }
 
+        /// <summary>
+        /// Constructor for BackupParser.
+        /// </summary>
+        /// <param name="backupChainPath">Folder path that contains sub folders of one full and multiple incremental backups.</param>
+        /// <param name="userLog">User configurable log object. </param>
+        /// <param name="workFolderPath">Folder pah for work folder. </param>
+        /// <param name="checkpointThreshold">Configurable checkpointThreshold. </param>
+        /// Pass an empty string if code package is not required for backup parsing. e.g. when backup has only primitive types.        
+        public BackupParser(string backupChainPath, ILog userLog, string workFolderPath, int checkpointThreshold = 200)
+        {
+            if (userLog != null)  log = userLog;
+            this.backupParserImpl = new BackupParserImpl(backupChainPath, String.Empty, userLog, workFolderPath, checkpointThreshold);
+        }
+
+        /// <summary>
+        /// Constructor for BackupParser.
+        /// </summary>
+        /// <param name="backupChainPath">Folder path that contains sub folders of one full and multiple incremental backups.</param>
+        /// <param name="codePackagePath">code package path.</param>
+        /// <param name="userLog">User configurable log object. </param>
+        /// <param name="workFolderPath">Folder path for work folder. </param>
+        /// <param name="checkpointThreshold">Configurable checkpointThreshold. </param>
+        /// Pass an empty string if code package is not required for backup parsing. e.g. when backup has only primitive types.        
+        public BackupParser(string backupChainPath, string codePackagePath, ILog userLog, string workFolderPath, int checkpointThreshold = 200)
```

Comment:
Instead of defining new constructors, can you use defaults and have just 1 constructor ?

Reply:
updated